pilot/pkg/serviceregistry/kube/controller/network_test.go

```diff
@@ -237,6 +237,58 @@ func TestSyncAllWorkloadsFromAmbient(t *testing.T) {
 
 	s := newAmbientTestServer(t, testC, "")
 
+	namespaceNotifyCh := make(chan string, 5)
+	defer close(namespaceNotifyCh)
+	s.controller.namespaces.AddEventHandler(controllers.ObjectHandler(func(o controllers.Object) {
+		namespaceNotifyCh <- o.GetName()
+	}))
+
+	waitNamespaceOrFail := func(t test.Failer, namespace string) {
```

Comment:
nit: this could be `assert.Equal(t, assert.ChannelHasItem(t, namespaceNotifyCh), namespace)`. Although that would fail if we want to ignore other namespace events.

Alternatively:
Make a assert.NewTracker
in the eventHandler: `tracker.Record()`
In the assert: tracker.WaitOrdered()